### PR TITLE
fix(refs T34297): Use Permission 'area_admin_boilerplates' in FE

### PR DESCRIPTION
**Pull request by salisdemos**
**Ticket:** https://yaits.demos-deutschland.de/T34297

If the Permission is dispabled, there shouldn't be a boilerplate modal/button in the editor

:warning:  Is the assumption correct, that a user, that can insert biolerplates can also create them? 

### How to review/test

Find a Project with "area_admin_boilerplates" enabled. Check, that You can use boilerplates like a charm. Then disable "area_admin_boilerplates" for Your current user and test if the boilerplates are gone without trowing any FE-Errors

e.g. as FPA `/verfahren/<procedure-id>/abwaegung/sview/<stn-id>`
and `/verfahren/testverfahren-flora/textbausteine`

:warning:  We may want to wait until codefreeze because its not necessarily required for this sprint and the changes are targeting several spots

### PR Checklist

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [ ] Link all relevant tickets
- [ ] Move the tickets on the board accordingly

### CODE REVIEW

File: client/js/bundles/statement/bulkEditStatement.js [hunk 1]
```diff
@@ -29,4 +29,8 @@ const components = {
   DpBulkEditStatement
 }
 
+if (hasPermission('area_admin_boilerplates')) {
+  stores.boilerplates = BoilerplatesStore
```
Comment (gruenbergerdemos) [inline client/js/bundles/statement/bulkEditStatement.js:33]: Remove the store above, too.

File: client/js/bundles/statement/assessmentTableView.js [hunk 2]
```diff
@@ -34,4 +33,8 @@ const components = {
   DpTable
 }
 
+if (hasPermission('area_admin_boilerplates')) {
```
Comment (gruenbergerdemos) [inline client/js/bundles/statement/assessmentTableView.js:36]: Do we need to import `hasPermission` here?
- If yes, the import is missing in the files below, too. 
- If not, can we remove the import from the files above?

**Review by gruenbergerdemos — CHANGES_REQUESTED** [review]

File: client/js/bundles/statement/bulkEditStatement.js (hunk 1 above)
Comment (salisdemos) [reply client/js/bundles/statement/bulkEditStatement.js:33]: fixed in f38170fcbc35bb136a7f6770a32d31c05451a7d2

File: client/js/bundles/statement/assessmentTableView.js (hunk 2 above)
Comment (salisdemos) [reply client/js/bundles/statement/assessmentTableView.js:37]: fixed in f38170fcbc35bb136a7f6770a32d31c05451a7d2

Comment (salisdemos) [reply client/js/bundles/statement/assessmentTableView.js:37]: It only has to be imported in entrypoints. In the vue-context its already available. I hope now I added all missing ones-

**Review by gruenbergerdemos — CHANGES_REQUESTED** [review]
One import is still missing

File: templates/bundles/DemosPlanCoreBundle/DemosPlanNews/news_admin_edit.html.twig [hunk 3]
```diff
@@ -250,24 +250,7 @@
...
-                <template v-slot:button>
-                    <button
-                        class="{{ 'menubar__button'|prefixClass }}"
-                        type="button"
-                        v-tooltip="'{{ 'boilerplate.insert'|trans }})'"
-                        @click.stop="$refs.boilerPlateModal.toggleModal()">
-                        <i class="{{ 'fa fa-puzzle-piece'|prefixClass }}"></i>
-                    </button>
-                </template>
+                {% include '@DemosPlanCore/DemosPlanNews/includes/news_editor_boilerplates_modal.html.twig' %}
```
Comment (gruenbergerdemos) [inline templates/bundles/DemosPlanCoreBundle/DemosPlanNews/news_admin_edit.html.twig:253]:
```suggestion
                {% include '@DemosPlanCore/DemosPlanNews/includes/news_editor_boilerplate_modal.html.twig' %}
```

File: templates/bundles/DemosPlanCoreBundle/DemosPlanNews/news_admin_new.html.twig [hunk 4]
```diff
@@ -210,24 +210,7 @@
...
-                <template v-slot:button>
-                    <button
-                        class="{{ 'menubar__button'|prefixClass }}"
-                        type="button"
-                        v-tooltip="'{{ 'boilerplate.insert'|trans }})'"
-                        @click.stop="$refs.boilerPlateModal.toggleModal()">
-                        <i class="{{ 'fa fa-puzzle-piece'|prefixClass }}"></i>
-                    </button>
-                </template>
+                {% include '@DemosPlanCore/DemosPlanNews/includes/news_editor_boilerplates_modal.html.twig' %}
```
Comment (gruenbergerdemos) [inline templates/bundles/DemosPlanCoreBundle/DemosPlanNews/news_admin_new.html.twig:213]:
```suggestion
                {% include '@DemosPlanCore/DemosPlanNews/includes/news_editor_boilerplate_modal.html.twig' %}
```

File: templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_unregistered_publicagency_email.html.twig [hunk 5]
```diff
@@ -105,23 +105,7 @@
...
-                    <template v-slot:button>
-                        <button
-                            class="{{ 'menubar__button'|prefixClass }}"
-                            type="button"
-                            v-tooltip="'{{ 'boilerplate.insert'|trans }})'"
-                            @click.stop="$refs.boilerPlateModal.toggleModal()">
-                            <i class="{{ 'fa fa-puzzle-piece'|prefixClass }}"></i>
-                        </button>
-                    </template>
+                    {% include '@DemosPlanCore/DemosPlanProcedure/includes/email_editor_boilerplates_modal.html.twig' %}
```
Comment (gruenbergerdemos) [inline templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_unregistered_publicagency_email.html.twig:108]:
```suggestion
                    {% include '@DemosPlanCore/DemosPlanProcedure/includes/email_editor_boilerplate_modal.html.twig' %}
```

**Review by gruenbergerdemos — CHANGES_REQUESTED** [review]

**Comment by salisdemos** [comment]
> One import is still missing

:cry:

**Comment by salisdemos** [comment]
> One import is still missing

here it is 600ac25683cf12436d9d35758a1258dd2672ad7c

**Comment by gruenbergerdemos** [comment]
You already added the missing import here [a9ce29d](https://github.com/demos-europe/demosplan-core/pull/1851/commits/a9ce29d750541911ca47fbe11919fc5638aee150) 

In vue we don't need to import `hasPermission`, I think? [600ac25](https://github.com/demos-europe/demosplan-core/pull/1851/commits/600ac25683cf12436d9d35758a1258dd2672ad7c) Otherwise it is also missing from the other vue files.

**Comment by salisdemos** [comment]
> You already added the missing import here [a9ce29d](https://github.com/demos-europe/demosplan-core/pull/1851/commits/a9ce29d750541911ca47fbe11919fc5638aee150)
> 

You lost me. from my perspective its all fine now, or did I still oversee some missing imports. 

> In vue we don't need to import `hasPermission`, I think? [600ac25](https://github.com/demos-europe/demosplan-core/pull/1851/commits/600ac25683cf12436d9d35758a1258dd2672ad7c) Otherwise it is also missing from the other vue files.'

Reverted in ebd3bc4f7872c131a0734b4edb43af82d8d0f8f7

**Comment by gruenbergerdemos** [comment]
Yes, it is all fine, sorry for the confusion. You added the missing import after my comment.

**Review by gruenbergerdemos — APPROVED** [review]
lgtm, thanks

**Pull request merged**